dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -160,12 +161,35 @@ private LinkageChecker(
         });
 
     // Filter classes in whitelist
-    SetMultimap<SymbolProblem, ClassFile> filteredMap = Multimaps
-        .filterValues(problemToClass.build(),
-            classFile -> !SOURCE_CLASSES_TO_SUPPRESS.contains(classFile.getClassName()));
+    SetMultimap<SymbolProblem, ClassFile> filteredMap =
+        Multimaps.filterEntries(problemToClass.build(), LinkageChecker::problemFilter);
     return ImmutableSetMultimap.copyOf(filteredMap);
   }
 
+  /**
+   * Returns true if the linkage error {@code entry} should be reported. False if it should be
+   * suppressed.
+   */
+  private static boolean problemFilter(Map.Entry<SymbolProblem, ClassFile> entry) {
```

Comment:
We have two different special cases here, that use different algorithms, but the same approach should work in both. I wonder if we should have a more general solution that lets us use configuration to whitelist instances we've verified are not problems.

Reply:
I like the idea of configuration. It’s going to be easy to explain special cases.